shim/src/realm.js

```diff
@@ -95,48 +72,43 @@ function createRealmRec(unsafeRec) {
   return realmRec;
 }
 
-function initRootRealm(selfClass, self, options) {
-  options = Object(options); // Todo: sanitize
-  // note: 'self' is the instance of the Realm, and 'selfClass' is the
-  // Realm constructor (facade) we build in buildChildRealm().
+function initRootRealm(parentUnsafeRec, self, options) {
+  options = Object(options); // todo: sanitize
 
   // In 'undefined' mode, intrinics are not provided, we create a root
   // realm using the fresh set of new intrinics from a new context.
 
   // todo: investigate attacks via Array.species
-  const newShims = options.shims || [];
-  const { allShims: oldShims } = getUnsafeRecForRealmClass(selfClass);
   // todo: this accepts newShims='string', but it should reject that
-  const allShims = arrayConcat(oldShims, newShims);
+  const { shims: newShims } = options;
+  const allShims = arrayConcat(parentUnsafeRec.allShims, newShims);
 
   // The unsafe record is returned with its constructors repaired.
   const unsafeRec = createNewUnsafeRec(allShims);
 
   // Define Realm onto new sharedGlobalDescs, so it can be copied onto the
-  // safeGlobal like the rest of the .
+  // safeGlobal like the rest of the globals.
   // eslint-disable-next-line no-use-before-define
-  const Realm = createRealmGlobalObject(unsafeRec);
-  registerUnsafeRecForRealmClass(Realm, unsafeRec);
+  createRealmGlobalObject(unsafeRec);
 
   const realmRec = createRealmRec(unsafeRec);
   registerRealmRecForRealmInstance(self, realmRec);
+
   // Now run all shims in the new RootRealm. We don't do this for
-  // compartments
+  // compartments.
+  const { safeEvalWhichTakesEndowments } = realmRec;
   for (const shim of allShims) {
-    // eslint-disable-next-line no-use-before-define
-    realmEvaluate(self, shim);
```

Comment:
In the pre-change version I see in the diff, it seems this is the only occurrence of `realmEvaluate` in this file. How can this be in scope? Where is the defining occurrence?

Reply:
Line 121
https://github.com/tc39/proposal-realms/pull/177/files#diff-467ff23f81f055b4ac27b6ec775d3381R121

Reply:
Weird. I don't know why I wasn't able to find it before. Thx.